### DISCUSSION TITLE
An implemention for a beatmaps detail option for touch input devices.

### DISCUSSION THREAD

**Discussion by cheeezie** [ppy/osu#21249] [Ideas] [answered]
From what I've seen while using a mouse, you can right click to get a 'Details' option that shows info and intel about a beatmap in its respective beatmap listing page. But on touch devices, either there isn't a way or a clear indication for such option.
![Screenshot_2022-11-16-01-56-56-15_25a1a32208bbcdc1d450b7aa854bb161](https://user-images.githubusercontent.com/102518950/202044602-c6e36b02-2633-4e6d-9d03-92522cfa728e.jpg)

**Comment by frenzibyte** [comment]
This will become possible once long tap is implemented, see https://github.com/ppy/osu-framework/issues/5200.

**Reply by cheeezie** [reply]
This holds true but the other options such as deleting and editing are available in the options button at the bottom of the UI, I was a little confused in why the details option wasn't there. Still, cheers.

**Comment by CenTdemeern1** [comment, marked as answer]
I just implemented this, and as I was about to propose it, I found this!

@frenzibyte I would personally prefer to have a button in this menu instead of having to long tap 😅 

https://drive.google.com/file/d/1XrcY7VE5aY2M8iGeKYM9ICT9LhCTghyK/view?usp=sharing
(video link)

**Reply by ghost** [reply]
Does your implementation keep existing key sequences, such as pressing `F3` then `5` to open the editor? Pretty sure the menu was made to match with stable.

**Reply by CenTdemeern1** [reply]
This is only a demonstration, not how I would do it for a more permanent implementation per se

No, it doesn't keep that key sequence, but for a more permanent implementation I'd probably put the button at the far right of the menu so it does

**Reply by peppy** [reply]
I think it's fine to add this. But it should use the same text as the context menu, not be arbitrarily different.

**Reply by CenTdemeern1** [reply]
Should it just say "details" then, with the second line empty?
My first version had the first line "Details", but the rest of the wording there was a bit awkward
![image](https://user-images.githubusercontent.com/24682280/204771714-2e969d5e-c965-4283-bf4d-881be1955745.png)
It wouldn't really fit this submenu to not have a second line of text

But I'm not against just having "Details" be the button's text.

Oh, and should I move the button to the right like I suggested?

**Reply by CenTdemeern1** [reply]
Moving the button to the right seems like it would cause a severe case of spaghetti code because the "Edit beatmap" button is being created in `PlaySongSelect.load`

**Reply by CenTdemeern1** [reply]
@peppy do you have any suggestions? Sorry if this mention is not warranted.

**Reply by peppy** [reply]
```csharp
View
details
```

should work fine.

It should be last, I'm sure the code can be coerced to look good.

**Reply by CenTdemeern1** [reply]
I think I'll just put the button in `PlaySongSelect.load` then, might as well. That'll deal with that issue.

**Reply by frenzibyte** [reply]
The button should display in all types of song select, not the "play" one only.

**Reply by CenTdemeern1** [reply]
`PlaySongSelect` adds its edit button after `SongSelect` though, and there is no method for inserting a button at a specified position (which could be used to insert the edit button before the details button)

**Reply by CenTdemeern1** [reply]
( #21497 )
